hail/src/test/scala/is/hail/annotations/UnsafeSuite.scala

```diff
@@ -397,6 +397,64 @@ class UnsafeSuite extends HailSuite {
     p.check()
   }
 
+  @Test def  testRegionAllocationSimple() {
+    using(new RegionPool) { pool =>
+      assert(pool.numFreeBlocks() == 0)
+      assert(pool.numRegions() == 0)
+      assert(pool.numFreeRegions() == 0)
+
+      val r = pool.getRegion(Region.REGULAR)
+
+      assert(pool.numRegions() == 1)
+      assert(pool.numFreeRegions() == 0)
+      assert(pool.numFreeBlocks() == 0)
+
+      r.clear()
+
+      assert(pool.numRegions() == 1)
+      assert(pool.numFreeRegions() == 0)
+      assert(pool.numFreeBlocks() == 0)
+
+      r.allocate(Region.SIZES(Region.REGULAR) - 1)
+      r.allocate(Region.SIZES(Region.REGULAR) - 1)
+      r.clear()
+
+      assert(pool.numRegions() == 1)
+      assert(pool.numFreeRegions() == 0)
+      assert(pool.numFreeBlocks() == 1)
```

Comment:
I'm not sure why we'd expect to see a free block appearing here---if we're allocating size-1 bytes into an empty region, we should see it allocating into an existing block in the first case, and grabbing a large chunk in the second. When it gets cleared, we should keep the current block and there shouldn't be any used blocks to return to the pool; the large chunk should get freed independently.